depends/toolchain.cmake.in

```diff
@@ -111,6 +109,15 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT CMAKE_HOST_APPLE)
   set(CMAKE_FRAMEWORK_PATH "@OSX_SDK@/System/Library/Frameworks")
 endif()
 
+
+# Customize pkg-config behaviour.
+cmake_path(APPEND CMAKE_FIND_ROOT_PATH "lib" "pkgconfig" OUTPUT_VARIABLE pkg_config_path)
+set(ENV{PKG_CONFIG_PATH} ${pkg_config_path})
```

Comment:
Why have these become env vars?

Reply:
They have always been:
- in the master branch: https://github.com/hebasto/bitcoin/blob/67c0d93982ad214f5e0c9509e9dc3d6d792ad97c/configure.ac#L22-L30
- in the current staging branch: https://github.com/hebasto/bitcoin/blob/f9b43f72e0d3723c56ffd83a7da52f6ed1980f67/cmake/module/CrossPkgConfig.cmake#L11-L12
- and in the pkg-config [documentation](https://manpages.debian.org/bullseye/pkg-config/pkg-config.1.en.html):
> `PKG_CONFIG_PATH`
> A colon-separated (on Windows, semicolon-separated) list of directories to search for .pc files. The default directory will always be searched after searching the path; the default is libdir/pkgconfig:datadir/pkgconfig where libdir is the libdir for pkg-config and datadir is the datadir for pkg-config when it was installed.
>
> `PKG_CONFIG_LIBDIR`
> Replaces the default `pkg-config` search directory, usually `/usr/lib/pkgconfig:/usr/share/pkgconfig`.

It's worth mentioning that [`set(ENV{<variable>} [<value>])`](https://cmake.org/cmake/help/latest/command/set.html#env) command:
> affects only the current CMake process, not the process from which CMake was called, nor the system environment at large, nor the environment of subsequent build or test processes.

Reply:
The `PKG_CONFIG_ARGN` sets a list of arguments to additionally pass to pkg-config. It is used to pass the `--static` option.

The `PKG_CONFIG_LIBDIR` and `PKG_CONFIG_PATH` affect the pkg-config behaviour in a way that is not achievable by other means according to pkg-config docs.

> that's to work around an autotools quirk

I can't see it as a workaround. Everything follows documentation.

Perhaps, I fail to grasp your idea. Maybe it is possible to put it in some (pseudo)code?